Issue/10

0hun opened the pull request:
ControllerAdvice 적용

0hun committed:
refactor : @EnableJpaAuditing main class 적용

@EnableJpaAuditing main class 적용
- @EnableJpaAuditing 적용하기 위해 빈 config 파일을 만드는 것 대신 main 클래스에서 적용하도록 수정

0hun committed:
feat (feature) : ExceptionController 추가

ExceptionController 추가
- @Controllerdvice를 적용하여 발생한 에러를 잡아 클라이언트에게 ResponseEntity 객체로 리턴하도록 추가

yhmane commented on src/main/java/com/delivery/DeliverServiceApplication.java:
```diff
@@ -2,7 +2,9 @@
 import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.data.jpa.repository.config.EnableJpaAuditing;
 
+@EnableJpaAuditing
```
변경된 코드는 jpaAuditingHandler 빈생성 오류를 유발합니다.

MockMVC 테스트시 해당 오류를 우회하는 여러가지 방안이 있습니다. 아래 내용을 참조하시면 무슨 말인지 이해가 되실거라 생각합니다.
https://stackoverflow.com/questions/41250177/getting-at-least-one-jpa-metamodel-must-be-present-with-webmvctest

코드 제출시 단위 테스트코드를 전부 수행하는게 좋을거 같아요 😅

0hun replied:
네네. 테스트나 기능이 변경되었을 경우 주의해야될 사항이 있다면 주석으로 부탁드릴게요.  해당 기능은 원래대로 복구해놓겠습니다.

yhmane commented on src/main/java/com/delivery/common/exception/ExceptionController.java:
```diff
@@ -0,0 +1,38 @@
+/**
+ *  @ControllerAdvice : 모든 컨트롤러에서 발생하는 예외를 잡아 처리해주는 어노테이션
+ */
+@ControllerAdvice
```
```suggestion
@RestControllerAdvice
```
를 이용하면 더 간결히 처리할 수 있을텐데 @ControllerAdvice를 이용한 이유가 있을까요?

0hun replied:
간결하다기보다는 RestControllerAdvice가 RestBody + ControllerAdvice를 포함하고 있는데요. 구현할 당시 RestControllerAdvice가 생각나지 않아, ControllerAdvice로 했습니다. ControllerAdvice로 바꾸면 될 것 같습니다.

yhmane commented on src/main/java/com/delivery/common/exception/ExceptionController.java:
```diff
@@ -0,0 +1,38 @@
+   * @param bindingResult
+   * @return ResponseEntity<List<ObjectError>>
+   */
+  @ExceptionHandler(MethodArgumentNotValidException.class)
```
👍

yhmane commented on src/test/java/com/delivery/user/controller/UserControllerTest.java:
```diff
@@ -45,7 +44,10 @@
-        mockMvc = MockMvcBuilders.standaloneSetup(userController).build();
+        mockMvc = MockMvcBuilders
+            .standaloneSetup(userController)
+            .setControllerAdvice(new ExceptionController())
```
👍

yhmane reviewed (changes requested):
잘 구현  해주셨으나 테스트시 오류가 있네요
테스트코드 Bean 주입도 중요한 부분이기에 Request Changes 하도록 하겠습니다 😅

0hun committed:
refactor : @EnableJpaAuditing config 추가

@EnableJpaAuditing config 추가
- 모든 테스트는 가장 기본이 되는 XXXApplication 클래스가 항상 로드되는데, @EnableJpaAuditing이 해당 클래스에 등록되어 있어서 모든 테스트들이 항상 JPA 관련 Bean들을 필요로 하고 있는 상태가 되기 때문에 모든 Bean들을 관리 하지 않기 위해
따로 Bean으로 등록

0hun committed:
refactor : @RestControllerAdvice 적용

@RestControllerAdvice 추가
- 결과 리턴시 Josn의 형태로 받아야 하기 때문에 RestBody로 전달해주는 RestControllerAdvice로 적용

yhmane reviewed (approved):
머지하겠습니다
